pkg/controller/jobframework/reconciler.go

```diff
@@ -583,6 +612,26 @@ func (r *JobReconciler) removeFinalizer(ctx context.Context, wl *kueue.Workload)
 func (r *JobReconciler) constructWorkload(ctx context.Context, job GenericJob, object client.Object) (*kueue.Workload, error) {
 	log := ctrl.LoggerFrom(ctx)
 
+	if cj, implements := job.(ComposableJob); implements {
+		wl, err := cj.ConstructWorkload(ctx, r.client, r.record)
+		if err != nil {
+			return nil, err
+		}
+
+		wl.Spec.PodSets = ResetMinCounts(wl.Spec.PodSets)
+
+		priorityClassName, source, p, err := r.extractPriority(ctx, wl.Spec.PodSets, job)
+		if err != nil {
+			return nil, err
+		}
+
+		wl.Spec.PriorityClassName = priorityClassName
+		wl.Spec.Priority = &p
+		wl.Spec.PriorityClassSource = source
+
+		return wl, nil
```

Comment:
This is common code.

Maybe `constructWorkload` should focus on getting the basics of the workload, and there could be another function `prepareWorkload` that adds the priority bits.

Reply:
Please address